pkg/rpcbackend/backend.go

```diff
@@ -141,11 +168,137 @@ func (rc *RPCClient) CallRPC(ctx context.Context, result interface{}, method str
 	return nil
 }
 
-// SyncRequest sends an individual RPC request to the backend (always over HTTP currently),
-// and waits synchronously for the response, or an error.
-//
-// In all return paths *including error paths* the RPCResponse is populated
-// so the caller has an RPC structure to send back to the front-end caller.
+type batchRequest struct {
+	rpcReq *RPCRequest
+	rpcRes chan *RPCResponse
+	rpcErr chan error
+}
+
+func (rc *RPCClient) startBatcher(ctx context.Context, batchDelay time.Duration, batchSize int) {
+	requestQueue := make(chan *batchRequest)
+
+	go func() {
+		ticker := time.NewTicker(batchDelay)
+		defer ticker.Stop()
+
+		var batch []*batchRequest
+
+		for {
+			select {
+			case req := <-requestQueue:
+				batch = append(batch, req)
+				if len(batch) >= batchSize {
+					rc.sendBatch(ctx, batch)
+					batch = nil
+				}
+			case <-ticker.C:
+				if len(batch) > 0 {
+					rc.sendBatch(ctx, batch)
+					batch = nil
+				}
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	rc.requestBatchQueue = requestQueue
+}
+
+func (rc *RPCClient) sendBatch(ctx context.Context, batch []*batchRequest) {
+	select {
+	case rc.requestBatchWorkerSlots <- true:
+		// wait for the worker slot and continue
+	case <-ctx.Done():
+		for _, req := range batch {
+			err := i18n.NewError(ctx, signermsgs.MsgRPCRequestBatchFailed)
+			req.rpcErr <- err
+		}
+		return
+	}
+	go func() {
```

Comment:
Suggest you split this into a second function called `sendBatch`, and the parent function is called `createWorker` - as it took me a while to understand the logic.